source/base/mpi.cc

```diff
@@ -778,11 +778,13 @@ namespace Utilities
 #ifdef DEAL_II_WITH_PETSC
 #  ifdef DEAL_II_WITH_SLEPC
       // Initialize SLEPc (with PETSc):
-      ierr = SlepcInitialize(&argc, &argv, nullptr, nullptr);
+      finalize_petscslepc = SlepcInitializeCalled ? false : true;
+      ierr                = SlepcInitialize(&argc, &argv, nullptr, nullptr);
```

Comment:
Why do you initialize here even though it is already initialized?

Reply:
Looks like it's idempotent, https://gitlab.com/petsc/petsc/-/blob/main/src/sys/objects/pinit.c#L1230-1236. Subsequent calls are ignored after `PetscInitialized` has been called once.

Reply:
yes